sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConfluentSchemaRegistryDeserializerProvider.java

```diff
@@ -91,7 +91,10 @@
             .put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, schemaRegistryUrl)
             .build();
     Deserializer<T> deserializer =
-        (Deserializer<T>) new KafkaAvroDeserializer(getSchemaRegistryClient());
+        (Deserializer<T>)
+            new ConfluentSchemaRegistryDeserializer(
+                getSchemaRegistryClient(),
+                new Schema.Parser().parse(getSchemaMetadata().getSchema()));
```

Comment:
This same line is used in the Coder part for a similar purpose, can we extract maybe this as a private method and reuse?
```
private Schema getAvroSchema() { return new Schema.Parser().parse(getSchemaMetadata().getSchema()); }
``